src/backends/cgpm_backend.py

```diff
@@ -922,6 +923,76 @@ def logpdf_joint(
             multiprocess=self._multiprocess,
         )
 
+    # Return python data object that's ready to be written in JSON syntax
+    def json_ready_models(self, bdb, population_name):
+        population_id = core.bayesdb_get_population(bdb, population_name)
+        generators = core.bayesdb_population_generators(
+            bdb, population_id)
+        assert len(generators) == 1
+        generator_id = generators[0]
+        stattypes = {name: stattype for (name, stattype) in
+                     bdb.sql_execute('''
+                         SELECT name, stattype FROM bayesdb_variable
+                         WHERE (generator_id IS NULL OR generator_id = ?)
+                         AND population_id = ?
+                     ''', (generator_id, population_id))}
+
+        categories = self._json_ready_categories(bdb, population_id, generator_id)
+
+        # Dict mapping colno to variable name
+        names = core.bayesdb_colno_to_variable_names(bdb, population_id, generator_id)
+        states = self._engine(bdb, generator_id).states
+        model_blobs = [self._json_ready_model(state, names) for state in states]
+
+        return {"column-statistical-types": stattypes,
+                "categories": categories,
+                "models": model_blobs}
+
+    def _json_ready_model(self, state, names):
+        column_groups = \
+            itertools.groupby(sorted(state.Zv().items()),
+                              key=operator.itemgetter(0))
```

Comment:
Whitespace: See HACKING for policy on hanging indents.

Reply:
HACKING says nothing about indentation

Reply:
Sorry, I meant [CONTRIBUTING.md](https://github.com/probcomp/bayeslite/blob/master/CONTRIBUTING.md)